src/doc/algo.cpp

```diff
@@ -60,6 +60,45 @@ void algo_line(int x1, int y1, int x2, int y2, void* data, AlgoPixel proc)
   }
 }
 
+/* Additional helper functions for the ellipse-drawing helper function
+   below corresponding to routines in Bresenham's algorithm. */
+namespace {
+  int bresenham_ellipse_error(int rx, int ry, int x, int y)
+  {
+    return x*x*ry*ry + y*y*rx*rx - rx*rx*ry*ry;
+  }
+
+  // Initialize positions x and y for Bresenham's algorithm
+  void bresenham_ellipse_init(int rx, int ry, int *px, int *py)
```

Comment:
Instead of using pointers here you can use references `(int rx, int ry, int& x, int& y)`